crates/egui/src/widgets/color_picker.rs

```diff
@@ -227,38 +227,146 @@ pub enum Alpha {
     BlendOrAdditive,
 }
 
-fn color_text_ui(ui: &mut Ui, color: impl Into<Color32>, alpha: Alpha) {
-    let color = color.into();
+fn color_text_ui(ui: &mut Ui, hsvag: &HsvaGamma, alpha: Alpha) {
+    use crate::style::ColorPickerInputType;
     ui.horizontal(|ui| {
-        let [r, g, b, a] = color.to_array();
+        let [r, g, b, a] = match ui.style().visuals.color_picker_input_values_type {
+            ColorPickerInputType::U8 => std::convert::Into::<Hsva>::into(*hsvag)
+                .to_srgba_unmultiplied()
+                .map(|c| c.to_string()),
+            ColorPickerInputType::F32 => {
+                let mut formatted_colors = (std::convert::Into::<Hsva>::into(*hsvag))
+                    .to_rgba_unmultiplied()
+                    .map(|c| {
+                        emath::round_to_decimals(c.into(), 3)
+                            .to_string()
+                            .char_range(0..5)
+                            .to_owned()
+                    });
+                for color in &mut formatted_colors {
+                    if color.contains('.') {
+                        *color = format!("{color:0<5}");
+                    } else if *color == "1" || *color == "0" {
+                        *color = format!("{color}.000");
+                    }
+                }
+                formatted_colors
+            }
+        };
 
-        if ui.button("📋").on_hover_text("Click to copy").clicked() {
-            if alpha == Alpha::Opaque {
-                ui.ctx().copy_text(format!("{r}, {g}, {b}"));
-            } else {
-                ui.ctx().copy_text(format!("{r}, {g}, {b}, {a}"));
+        ui.horizontal(|ui| {
+            if ui
+                .button(
+                    ui.style()
+                        .visuals
+                        .color_picker_input_values_type
+                        .to_string(),
+                )
+                .clicked()
+            {
+                let mut visuals = ui.style().visuals.clone();
+                match ui.style().visuals.color_picker_input_values_type {
+                    ColorPickerInputType::U8 => {
+                        visuals.color_picker_input_values_type = ColorPickerInputType::F32;
+                    }
+                    ColorPickerInputType::F32 => {
+                        visuals.color_picker_input_values_type = ColorPickerInputType::U8;
+                    }
+                }
+                ui.ctx().set_visuals(visuals);
             }
-        }
+            if ui.button("📋").on_hover_text("Click to copy").clicked() {
+                if alpha == Alpha::Opaque {
+                    ui.ctx().copy_text(format!("{r}, {g}, {b}"));
+                } else {
+                    ui.ctx().copy_text(format!("{r}, {g}, {b}, {a}"));
+                }
+            }
```

Comment:
I think we should move this copy-button before the new RGB-buttons. Right now we show the RGB values twice - first as buttons, and then as labels. That is redundant and redundant.

Reply:
Rearranged the buttons layout, removed the label.